webservice/webservice.py

```diff
@@ -322,7 +339,85 @@ def get_manifest():
     # Get the response back
     logger.info("Creating the API response")
     response = es_td.transform_manifest(filters=filters)
-    # Return the excel file
+    if format == 'tsv':  # returns metadata as TSV file
+        return response
+    elif format == 'bdbag':  # returns presigned URL to a BDBag in S3
+        response_bag = bdbag_response(response)
+        if 'status' in response_bag:
+            http_status = response_bag['status']
+        else:
+            http_status = 200
+        return jsonify(response_bag), http_status
+
+
+def bdbag_response(response_obj):
+    """
+    Create a response for BDBag upload to S3. Returns just the presigned URL of
+    the S3 location if all goes well. Otherwise the HTTP status code, error
+    code, and a message are returned in JSON format.
+    
+    :param response_obj: Contains the selected metadata as a TSV.
+    :type response_obj: A Flask response object.
+    :return response: Information depends on status of actions.
+    :rtype response: JSON
+    """
+    logger = logging.getLogger('dashboardService.webservice.bdbag_response')
+    # Create and return the BDbag folder.
+    bag_name = 'manifest'
+    bag_info = {'organization': '',
+                'data_type': '',
+                'date_created': datetime.datetime.now().isoformat()}
+
+    # Instantiate bag object.
+    bag = BagHandler(data=response_obj.get_data(),
+                     bag_info=bag_info,
+                     bag_name=bag_name)
+    zipped_bag = bag.create_bag()  # path to compressed bag
+    logger.info("Creating a compressed BDbag containing manifest.")
+
+    # Import bucket environment variable, launch instance of S3-file handler,
+    # and upload the BDBag file to S3.
+
+    # Transfer parameters.
+    azul_s3_aws_region = os.getenv("AZUL_S3_AWS_REGION", 'us-west-2')
+    azul_presigned_url_expiration = \
+        os.getenv("AZUL_PRESIGNED_URL_EXPIRATION", 3600)  # in seconds
+    bucket_key = str(uuid.uuid4())
+    azul_s3_bucket = os.getenv("AZUL_S3_BUCKET")
```

Comment:
It might be best to have default values for this and the next two. Not that the default values for these will work, like the previous two, but even so, the failures might be more obvious than if you have None as the values

Reply:
Done.